pulpcore/cli/core/task.py

```diff
@@ -112,3 +114,33 @@ def cancel(
             if not re.match("Task /pulp/api/v3/tasks/[-0-9a-f]*/ canceled", str(e)):
                 raise e
         click.echo(_("Done."), err=True)
+
+
+DATETIME_FORMATS = ["%Y-%m-%d", "%Y-%m-%dT%H:%M:%S"]
+
+
+@task.command()
+@click.option(
+    "--finished-before",
+    "finished",
+    help=_("Purge task-records whose 'finished' time is **before** the time specified."),
+    type=click.DateTime(formats=DATETIME_FORMATS),
+)
+@click.option(
+    "--states",
+    help=_("Only purge tasks in the specified state(s)."),
+    type=click.Choice(["skipped", "completed", "failed", "canceled"]),
+    multiple=True,
+)
+@pass_entity_context
+@pass_pulp_context
+def purge(
+    pulp_ctx: PulpContext,
+    task_ctx: PulpTaskContext,
+    finished: Optional[datetime],
+    states: Optional[List[str]],
+) -> None:
+    pulp_ctx.needs_plugin(PluginRequirement("core", "3.17.0.dev"))
+    state_list = list(states) if states else None
```

Comment:
According to the typing info, states is already a List[str] or None.
But i think, click passes it as a Tuple. So either this transformation is unnecessary, or the typing info is incorrect (and yes, sadly mypy does not account for the click decorators that rigorously).